### PR TITLE
Fix the bug of reading environment variables

### PR DESCRIPTION
**Pull request by xiaoxiaoafeifei**
Hello，when I read the code, I found a bug when the **read_afl_environment** function reads environment variables

### CODE REVIEW

File: src/afl-fuzz-state.c [hunk 1]
```diff
@@ -164,229 +164,227 @@ void afl_state_init(afl_state_t *afl, uint32_t map_size) {
 
 void read_afl_environment(afl_state_t *afl, char **envp) {
 
-  int   index = 0, issue_detected = 0;
+  int index = 0, issue_detected = 0;
   char *env;
   while ((env = envp[index++]) != NULL) {
 
-    if (strncmp(env, "ALF_", 4) == 0) {
+    if (strncmp(env, "ALF_", 4) != 0) {
```
Comment (vanhauser-thc) [inline src/afl-fuzz-state.c:171]: you inverse this. why? this makes it all wrong

**Comment by vanhauser-thc** [comment]
I think you misinterpreted the code.
some people write AFL_DEBUG when they mean AFL_DEBUG and this is what the function warns about. your change removes that feature.

File: src/afl-fuzz-state.c (hunk 1 above)
Comment (xiaoxiaoafeifei) [reply src/afl-fuzz-state.c:171]: void read_afl_environment(afl_state_t *afl, char **envp) {

  int   index = 0, issue_detected = 0;
  char *env;
  while ((env = envp[index++]) != NULL) {

    if (strncmp(env, "ALF_", 4) == 0) {

      WARNF("Potentially mistyped AFL environment variable: %s", env);
      issue_detected = 1;

    } else if (strncmp(env, "AFL_", 4) == 0) {


I have a doubt， what is the meaning of the same condition as if and else branch？

**Comment by xiaoxiaoafeifei** [comment]
> I think you misinterpreted the code.
> some people write AFL_DEBUG when they mean AFL_DEBUG and this is what the function warns about. your change removes that feature.

void read_afl_environment(afl_state_t *afl, char **envp) {

int index = 0, issue_detected = 0;
char *env;
while ((env = envp[index++]) != NULL) {

**if (strncmp(env, "ALF_", 4) == 0) {**

  WARNF("Potentially mistyped AFL environment variable: %s", env);
  issue_detected = 1;

**} else if (strncmp(env, "AFL_", 4) == 0) {**
**I have a doubt， what is the meaning of the same condition as if and else branch？**

File: src/afl-fuzz-state.c (hunk 1 above)
Comment (vanhauser-thc) [reply src/afl-fuzz-state.c:171]: look closely, it is not the same. one is aLf, the other aFl :)

Comment (xiaoxiaoafeifei) [reply src/afl-fuzz-state.c:171]: > look closely, it is not the same
Ok, thanks for your reply

**Pull request closed without merging**